File: app/src/main/java/com/neworin/easyweather/module/search/presenter/SearchPresenter.java
package com.neworin.easyweather.module.search.presenter;

import android.content.Context;

import com.neworin.easyweather.entity.H5Weather;
import com.neworin.easyweather.module.home.model.HomeModelImpl;
import com.neworin.easyweather.module.home.model.IHomeModel;
import com.neworin.easyweather.module.home.view.ISearchView;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

/**
 * Author: NewOrin Zhang
 * Email: neworin@163.com
 * Description:
 */
public class SearchPresenter {

    private ISearchView mHomeView;
    private IHomeModel mHomeModel;
    private Context mContext;

    public SearchPresenter(ISearchView view, Context context) {
        this.mHomeView = view;
        this.mHomeModel = new HomeModelImpl();
        this.mContext = context;
    }

    public void getCityWeather(String city) {
        mHomeModel.getWeatherByCityName(city, new Callback<H5Weather>() {
            @Override
            public void onResponse(Call<H5Weather> call, Response<H5Weather> response) {
                mHomeView.refreshData(response.body().HeWeather5);
            }

            @Override
            public void onFailure(Call<H5Weather> call, Throwable t) {
                t.printStackTrace();
            }
        });
    }

    public void searchCity(String city) {
        mHomeModel.searchCity(city, new Callback<H5Weather>() {
            @Override
            public void onResponse(Call<H5Weather> call, Response<H5Weather> response) {

            }

            @Override
            public void onFailure(Call<H5Weather> call, Throwable t) {

            }
        });
    }
}

File: app/src/main/java/com/neworin/easyweather/module/search/presenter/SearchPresenter.kt
package com.neworin.easyweather.module.search.presenter

import android.content.Context
import android.content.SharedPreferences
import com.alibaba.fastjson.JSON
import com.neworin.easyweather.db.utils.DBManager
import com.neworin.easyweather.entity.Basic

import com.neworin.easyweather.entity.H5Weather
import com.neworin.easyweather.entity.db.Citys
import com.neworin.easyweather.module.home.model.HomeModelImpl
import com.neworin.easyweather.module.home.model.IHomeModel
import com.neworin.easyweather.module.home.view.ISearchView
import com.neworin.easyweather.module.search.model.SearchModel
import com.neworin.easyweather.module.search.model.SearchModelImpl
import com.neworin.easyweather.utils.Constant
import com.neworin.easyweather.utils.SharedPreferenceUtil
import com.orhanobut.logger.Logger
import io.reactivex.Observer
import io.reactivex.disposables.Disposable

import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

/**
 * Author: NewOrin Zhang
 * Email: neworin@163.com
 * Description:
 */
class SearchPresenter(private val mSearchView: ISearchView, private val mContext: Context) {

    private val mSearchModel: SearchModel

    init {
        this.mSearchModel = SearchModelImpl()
    }

    fun searchCity(city: String) {
        val dbManager = DBManager(mContext)
        dbManager.doQueryCity(city, object : Observer<List<Citys>> {
            override fun onSubscribe(d: Disposable) {

            }

            override fun onNext(citys: List<Citys>) {
//                provincesList = citys
            }

            override fun onError(e: Throwable) {
                Logger.d("get data error")
            }

            override fun onComplete() {
//                Logger.d("size = " + provincesList.size)
            }
        })
    }

    fun insertSharedCity(basic: Basic) {
        val cities = SharedPreferenceUtil.getString(mContext, Constant.SHARED_CITIES_TAG)
        var basicList = JSON.parseArray(cities, Basic::class.java)
        if (null == basicList) {
            basicList = ArrayList()
        }
        basicList.add(0, basic)
        SharedPreferenceUtil.putString(mContext, Constant.SHARED_CITIES_TAG, JSON.toJSONString(basicList))
    }
}